modules/apps/dynamic-data-mapping/dynamic-data-mapping-taglib/src/main/resources/META-INF/resources/html/start.jsp

```diff
@@ -198,16 +198,21 @@
 				});
 			};
 
-			var onDestroyPortlet = function (event) {
-				if (event.portletId === '<%= portletDisplay.getId() %>') {
-					liferayDDMForm.destroy();
+			// Again, note the somewhat icky use of the `__unsafe__` namespace
+			// here.
 
-					Liferay.detach('inputLocalized:localeChanged', onLocaleChange);
-					Liferay.detach('destroyPortlet', onDestroyPortlet);
-				}
-			};
+			var {dispose} = Liferay.State.__unsafe__.subscribe(
+				'frontend-js-web:active-portlet-ids',
+				(ids) => {
+					if ('<%= portletDisplay.getId() %>' in ids) {
```

Comment:
This should of cause be `if (!('<%= portletDisplay.getId() %>' in ids) {`, suggesting that this is any icky way to model lifecycle that is hard to interact with (ie. easy to get wrong).

This is what I was getting at when I said a piecemeal event-by-event migration might not make a lot of sense. Instead of a list of `activeIds`, we could have a richer representation that modeled which point of the lifecycle each given portlet is in, with those points roughly corresponding to the events under the old system:

```
Liferay.fire('addPortlet', {
Liferay.fire('allPortletsReady', {
Liferay.fire('closePortlet', options);
Liferay.fire('destroyPortlet', options);
Liferay.fire('initLayout');
Liferay.fire('portletReady', {
Liferay.fire('updatedLayout');
Liferay.fire(portlet.portletId + ':portletRefreshed',
```

The other thing that smells to me a bit about modeling this as global state is it is really only ever a two-way conversation between a `master` (`liferay/portlet.js`) and a specific piece of code in a JSP. That is, the portlets never care about each other, they only ever care about their own state. So it seems odd to want the state of _all_ portlets to be globally visible to _everyone_. (Note that the existing event-based model does leak global state to anybody who listens, but it does so implicitly, and only to people who are listening when the events happen; in contrast, reifying this as global state means that anybody can drop in at any time and see the "big picture" at that instant even without having been present when the events happened... that _sounds_ like it could theoretically be useful, but it also sets of my YAGNI alarm.)